src/App.tsx

```diff
@@ -43,28 +38,24 @@ export default function App(): JSX.Element {
     const authTokensFromQuery: OptionalAuthTokens = parseAuthParams(
       window.location.search
     );
-    const storedAuthTokens: OptionalAuthTokens = getTokens();
 
-    if (storedAuthTokens) {
-      removeTokens();
+    if (authTokensFromQuery) {
+      storeTokens(authTokensFromQuery);
     }
 
-    if (authTokensFromQuery) {
-      api
-        .testAuthCredentials(authTokensFromQuery)
-        .then(() => {
-          saveAuthTokes(authTokensFromQuery);
-          setLoading(false);
-        })
-        .catch((e: Error) => onAuthFail(e.message));
-    } else if (storedAuthTokens) {
+    const storedAuthTokens: OptionalAuthTokens = getTokens();
```

Comment:
Do we want to always get the tokens from localstorage, even if we already have access to the tokens from query above? I guess it makes the code simpler, but also now we need to always access (get) local storage even we would not need to. But I guess it is a good way to test that the _storeTokens_ part worked?